Refactor Updates and Examples for 6.6.1 Advanced Bucket Accessors

jon-strabala opened the pull request:
This PR is specific for 6.6.1 

1. Refactors the navigation into hierarchical sections WRT Eventing Examples and Scriptlets
2. Refactors the Advanced Bucker Accessors into its own page
3. Add nine (9) new Scriptlet examples
4. Other minor updates.

jon-strabala committed:
Refactor "Advanced Bucket Accessors" to new page

1. Call "Bucket Accessors" "Basic Bucket Accessors" to differentiate from "Advanced Bucket Accessors"
2. Refactor "Advanced Bucket Accessors" details to a seperate page like Timers and cURL with links to all seven items for details.

jon-strabala committed:
Refactor "Advanced Bucket Accessors" to new page

1. Call "Bucket Accessors" "Basic Bucket Accessors" to differentiate from "Advanced Bucket Accessors"
2. Refactor "Advanced Bucket Accessors" details to a seperate page like Timers and cURL
3. Add not about performance and ease of use for "Basic Bucket Accessors" v. "Advanced Bucket Accessors"

jon-strabala committed:
Since in a new page unindent most items

jon-strabala committed:
Add new page "Advanced Bucket Accessors"

jon-strabala committed:
Minor formatting

jon-strabala committed:
bad link in main nav

jon-strabala committed:
Make nav to scriplets hierarchical

jon-strabala committed:
Add new hrefs for nested scriptlet nav sections

jon-strabala committed:
Add scriplet section titles

jon-strabala committed:
Add title and highlight step by step

jon-strabala committed:
Add N1QL TTL option available since 6.5.1

jon-strabala committed:
needs to go into bucket source

jon-strabala committed:
add anchor for eventing

jon-strabala committed:
use anchor in N1QL change

jon-strabala committed:
Clean up links to new page update log descriptions

Clean up links to new page, better standards
Enhance system log descriptions both application and system

jon-strabala committed:
add anchors for system and application logs

jon-strabala committed:
fix some links

alter _ to - in som new / recent external hrefs

jon-strabala committed:
update anchors

update _ to - in new / recent anhors

jon-strabala committed:
spelling fixes

jon-strabala committed:
spelling fixes

jon-strabala committed:
new intern anchor "Added Language Features"

jon-strabala committed:
Add anchors

jon-strabala committed:
Introduce and link sub pages

The createTimer() function call
The cancelTimer() function call
The curl() function call

jon-strabala committed:
small format fix

jon-strabala committed:
camelcase fixes

jon-strabala committed:
camelcase fixes

jon-strabala committed:
formatting tab name

jon-strabala committed:
Add tab for N1QL supported since 6.5.1

jon-strabala committed:
Syntax and text

jon-strabala committed:
Add limitation on what can be passed to a context

jon-strabala committed:
fix typos

jon-strabala committed:
Remove 6.5.1 this is a 6.6+ doc set

jon-strabala committed:
Remove 6.5.1 this is a 6.6+ doc set

jon-strabala committed:
test

jon-strabala committed:
typo

jon-strabala committed:
test2

jon-strabala committed:
remove test

I wanted to ensure good formatting for a 6.5 document update - it worked so reverting the TAG
 [.status]#Couchbase Server 6.5.1#::

jon-strabala committed:
Add 6.6.1 CAS variant

UPdate scriptlet with 6.6.1 CAS variant

jon-strabala committed:
layout fix and spelling fixes

jon-strabala committed:
Add 6.6.1 tags

jon-strabala committed:
small clarity fix

jon-strabala committed:
Add more info on behavior

jon-strabala committed:
Add link for adv. GET/INSERT/REPLACE example

Scriptlet: Keep the Last N User Items
xref:eventing-handler-keepLastN.adoc[Keep the Last N User Items].

jon-strabala committed:
Update eventing-examples.adoc

jon-strabala committed:
Add eventing scriptlet example

eventing-handler-keepLastN_661

jon-strabala committed:
New Advanced Accessors section

jon-strabala committed:
Split out and make new Scriptlet for 6.6.1 

Originally I combined the two versions, changed my mind

jon-strabala committed:
Split out the 6.6.1 version

Originally I combined the two versions, changed my mind

jon-strabala committed:
Fix ref due to split of Scriptlet

jon-strabala committed:
move block alter name

jon-strabala committed:
rename for consistancy

jon-strabala committed:
Updates to ref improved 6.6.1 Function

Updates to ref improved 6.6.1 Function docControlledSelfExpiry_661

jon-strabala committed:
New Function docControlledSelfExpiry_661

jon-strabala committed:
Typo

jon-strabala committed:
New example TTL adv.bucket ops

jon-strabala committed:
Add docControlledSelfExpiry_661 Scriplet

jon-strabala committed:
move page into pages

jon-strabala committed:
link to TTL scriplet

jon-strabala committed:
update text

jon-strabala committed:
syntax

jon-strabala committed:
move scriptlet block

jon-strabala committed:
minor update to Advanced GET to match Scriptlet

jon-strabala committed:
Add advancedGetOp

jon-strabala committed:
add advancedGetOp

jon-strabala committed:
add advancedGetOp

jon-strabala committed:
advancedGetOp in wrong block

jon-strabala committed:
Update eventing-examples.adoc

jon-strabala committed:
rename

jon-strabala committed:
format JSON output for legibility

jon-strabala committed:
format final JSON output for consistency

jon-strabala committed:
clarity text

jon-strabala committed:
make consistent with advancedInsertOp scriptlet

jon-strabala committed:
New Scriptlet advancedInsertOp

jon-strabala committed:
Add Scriptlet advancedInsertOp

jon-strabala committed:
Add Scriptlet advancedInsertOp

jon-strabala committed:
formating

jon-strabala committed:
formatting

jon-strabala committed:
clarity

jon-strabala committed:
formatting

jon-strabala committed:
update descriptive text

jon-strabala committed:
formatting

jon-strabala committed:
formatting and break up log messages

jon-strabala committed:
Initial work on advancedUpsertOp

jon-strabala committed:
updates to Advanced GET and Advanced INSERT

More description on Advanced GET, explaining that unlic Basic GET you will get CAS
update Advanced INSERT to match the scriptlet better.

jon-strabala committed:
formatting of log output

jon-strabala committed:
spelling fixes

jon-strabala committed:
Can not "bold" in a code block

jon-strabala committed:
Improve by separating mutations

jon-strabala committed:
Add xref to advancedUpsertOp

jon-strabala committed:
xref for advancedUpsertOp

jon-strabala committed:
fix xref to main doc

jon-strabala committed:
fix xref to main doc

jon-strabala committed:
improved GET description

jon-strabala committed:
Update Advanced REPLACE to be more like Scriptlet

jon-strabala committed:
Make REPLACE function more robust for testing

jon-strabala committed:
Add Scriptlet advancedReplaceOp

jon-strabala committed:
Add xref fro advancedReplaceOp

jon-strabala committed:
Add xref for advancedReplaceOp Scriptlet

jon-strabala committed:
Make similar to advancedDeleteOp

jon-strabala committed:
fix typo

jon-strabala committed:
Add advancedDeleteOp Scriptlet

jon-strabala committed:
remover unused arg new_doc from Delete

jon-strabala committed:
xref for advancedDeleteOp Scriptlet

jon-strabala committed:
xref for advancedDeleteOp Scriptlet

jon-strabala committed:
Add advancedIncrementOp Scriptlet

jon-strabala committed:
INCREMENT and DECREMENT to be like Scriptlets

jon-strabala committed:
Add advancedDecrementOp Scriptlet

jon-strabala committed:
put in a pre-filter

jon-strabala committed:
Xref advancedIncrementOp and advancedDecrementOp

jon-strabala committed:
Xref advancedIncrementOp and advancedDecrementOp

New Scriptlets

jon-strabala committed:
One general purpose link to the needed examples

jon-strabala committed:
Update the GET information

jon-strabala committed:
gammer

jon-strabala committed:
formatting

jon-strabala committed:
bad tab display

jon-strabala committed:
bad tab display

jon-strabala committed:
bad tab formatting

jon-strabala committed:
bad tab formatting

jon-strabala committed:
bad tab formatting

jon-strabala committed:
bad tab formatting

jon-strabala committed:
direction of count

jon-strabala committed:
spelling and clarity

jon-strabala committed:
expiry_date NOT expiry_data

jon-strabala committed:
diction

jon-strabala committed:
diction

jon-strabala committed:
diction

jon-strabala committed:
diction

jon-strabala committed:
diction

jon-strabala committed:
diction

jon-strabala committed:
diction

jon-strabala committed:
Improve recurring_timer example

Best practice re-arm timer first before any work to make sure a timeout or error doesn't stop the recurrence.
Update the number of documents in the metadata bucket as timers change the counts after 6.6.1 to 1280 down form 2048
Due to slight code JavaScript change an updated recurring_timer_03_editor_with_code.png is needed

jon-strabala committed:
Code change requires new image

Replace recurring_timer_03_editor_with_code.png

jon-strabala committed:
Updates for clarity and also future blogging

amarantha-k committed:
minor editorial update and fix typo

amarantha-k commented on modules/eventing/pages/eventing-handler-docControlledSelfExpiry_661.adoc:
```diff
@@ -0,0 +1,160 @@
+= Function: Document Controlled Expiry 6.6.1
```
Rename the file to remove the version number (eventing-handler-advanced-docControlledSelfExpiry.adoc)

jon-strabala replied:
There are two versions (one will work up through 6.6.0 and the other will work starting from 6.6.1 on)
eventing-handler-docControlledSelfExpiry.adoc
eventing-handler-docControlledSelfExpiry_661.adoc
I have no issues with renaming - but we still need the two versions since this is a 6.6.0 document set.

amarantha-k commented on modules/eventing/pages/eventing-handler-keepLastN_661.adoc:
```diff
@@ -0,0 +1,240 @@
+= Function: Keep the Last N User Items 6.6.1
```
Rename the file to remove the version number (eventing-handler-advanced-keepLastN.adoc)

jon-strabala replied:
There are two versions (one will work up through 6.6.0 and the other will work starting from 6.6.1 on)
eventing-handler-keepLastN.adoc
eventing-handler-keepLastN_661.adoc
I have no issues with renaming - but we still need the two versions since this is a 6.6.0 document set.

amarantha-k commented on modules/eventing/pages/eventing-handler-keepLastN_661.adoc:
```diff
@@ -0,0 +1,240 @@
+* xref:eventing-handler-keepLastN.adoc[keepLastN (userspace CAS)]
+* <<keepLastN_661,keepLastN_661 (true CAS)>>
+
+[#keepLastN_661]
```
rename to [#advanced_keepLastN] so the URL stay the same for future releases as well

jon-strabala replied:
There are two versions (one will work up through 6.6.0 and the other will work starting from 6.6.1 on)
eventing-handler-keepLastN.adoc
eventing-handler-keepLastN_661.adoc
I have no issues with renaming - but we still need the two versions since this is a 6.6.0 document set.

amarantha-k reviewed (changes requested):
@jon-strabala Need to update the file names and the respective xrefs in the nav file.

Commit:
Rename files to remove version number in filename and updated related links

jon-strabala committed:
Merge pull request #1 from amarantha-k/jon-strabala-release/6.6

Rename files to remove version number in filename and updated related…

jon-strabala committed:
Complete rename to remove _661

jon-strabala committed:
Complete rename to remove _661

jon-strabala committed:
Complete rename to remove _661

jon-strabala committed:
Complete rename to remove _661

jon-strabala committed:
Make names consistent

Make names consistent - I don't like spaces in my function names (in fact it is illegal)

jon-strabala committed:
Make names consistent (eventing functions)

Make names consistent - I don't like spaces in my function names (in fact it is illegal)

jon-strabala committed:
Add hyphen to allow a word break

This makes the main NAV more pleasing ...

WAS:

advancedDocControlledSelfExpir
y

NOW:

advanced-
DocControlledSelfExpiry

amarantha-k reviewed (approved):
Thanks for incorporating the feedback. lgtm
